freud/box.pyx

```diff
@@ -223,26 +225,28 @@ cdef class Box:
                 Vectors of real coordinates: :math:`\left(3\right)` or :math:`\left(N, 3\right)`.
         """  # noqa: E501
         fractions = np.asarray(fractions)
-        if fractions.ndim > 2 or fractions.shape[fractions.ndim-1] != 3:
-            raise ValueError(
-                "Invalid dimensions for fractions given to makeCoordinates. "
-                "Valid input is an array of shape (3,) or (N,3).")
-
-        fractions = freud.common.convert_array(fractions)
+        flatten = fractions.ndim == 1
+        fractions = np.atleast_2d(fractions)
+        fractions = freud.common.convert_array(fractions, shape=(None, 3))
 
-        if fractions.ndim == 1:
-            fractions[:] = self._makeCoordinates(fractions)
-        elif fractions.ndim == 2:
-            for i, f in enumerate(fractions):
-                fractions[i] = self._makeCoordinates(f)
-        return fractions
+        for i, f in enumerate(fractions):
+            fractions[i] = self._makeCoordinates(f)
+        return np.squeeze(fractions) if flatten else fractions
 
     def _makeCoordinates(self, f):
         cdef const float[::1] l_vec = f
         cdef vec3[float] result = self.thisptr.makeCoordinates(
             <const vec3[float]&> l_vec[0])
         return [result.x, result.y, result.z]
 
+    def _makeCoordinates_many(self, vecs):
+        cdef const float[:, ::1] l_points = vecs
+        cdef unsigned int Np = l_points.shape[0]
+        with nogil:
```

Comment:
I _think_ this automatically drops the GIL since the method is marked `nogil`. Please verify this in the Cython docs and then remove with `with nogil:` context manager.

Reply:
I believe marking a function as `nogil` just indicates that it's safe to be called while the gil GIL is released, but does not release the GIL. However, this function is also not currently marked as `nogil`, unless I missed some history?

Reply:
According to Cython,
> Note The nogil function annotation declares that it is safe to call the function without the GIL. It is perfectly allowed to execute it while holding the GIL. The function does not in itself release the GIL if it is held by the caller.
so, I think we need `nogil`.